File: .threads/ponger.py
from threading import Thread
from time import sleep

class Ponger(Thread):
    def __init__(self):
        super(Ponger, self).__init__()
        self.daemon = True
        self.__killed = False

    def run(self):
        i = 1
        while not self.__killed:
            print(f"{i}. pong")
            sleep(2)
            i += 1

    def kill(self):
        self.__killed = True
        print("ponger killed")

File: .threads/Ponger.java
import java.lang.Runnable;
import java.lang.Exception;

public class Ponger extends Thread {
    private boolean killed = false;

    public Ponger() {
    }

    @Override
    public void run() {
        try {
            int i = 1;
            while(!killed) {
                System.out.println(i + ". pong");
                Thread.sleep(2000);
                i++;
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public void kill() {
        killed = true;
        System.out.println("Ponger killed");
    }
}

